src/DocumentFormat.OpenXml.Framework/MarkupCompatibilityAttributes.cs

```diff
@@ -42,5 +44,50 @@ public class MarkupCompatibilityAttributes
         ///  a set of namespace names.
         /// </summary>
         public StringValue? MustUnderstand { get; set; }
+
+        /// <inheritdoc/>
+        public override bool Equals(object? obj)
+        {
+            return this.Equals(obj as MarkupCompatibilityAttributes);
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(MarkupCompatibilityAttributes? other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (object.ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return (this.Ignorable == null == (other.Ignorable == null)) &&
```

Comment:
Could this use pattern matching to simplify it?

Reply:
I'm not sure how you would like to use pattern matching to make this simpler.

I'm pretty used to this, so it seems easy to me, but if you have a better way, please expand on it.